quickwit-cluster/src/cluster.rs

```diff
@@ -307,6 +351,96 @@ mod tests {
         tmp_dir.close().unwrap();
     }
 
+    #[tokio::test]
+    async fn test_cluster_single_node() -> anyhow::Result<()> {
+        let tmp_dir = tempfile::tempdir()?;
+        let host_key_path = tmp_dir.path().join("host_key");
+        let host_key = read_host_key(host_key_path.as_path())?;
+
+        let tmp_port = available_port()?;
+        let addr_string = format!("127.0.0.1:{}", tmp_port);
+        let remote_host: SocketAddr = addr_string.as_str().parse()?;
+
+        let cluster = Arc::new(EpidemicCluster::new(host_key, remote_host)?);
+
+        let members = cluster.members();
+        let expected = vec![Member {
+            host_key,
+            listen_addr: remote_host,
+            is_self: true,
+        }];
+        assert_eq!(members, expected);
+
+        cluster.leave();
+
+        tmp_dir.close()?;
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_cluster_multiple_node() -> anyhow::Result<()> {
+        let tmp_dir = tempfile::tempdir()?;
+
+        let host_key_path1 = tmp_dir.path().join("host_key1");
+        let host_key1 = read_host_key(host_key_path1.as_path())?;
+        let tmp_port1 = available_port()?;
+        let addr_str1 = format!("127.0.0.1:{}", tmp_port1);
+        let remote_host1: SocketAddr = addr_str1.as_str().parse()?;
+        let cluster1 = EpidemicCluster::new(host_key1, remote_host1)?;
```

Comment:
this should be moved to a helper function

Reply:
I changed the code a bit. Unnecessary variables have been removed.
It's not a helper function, but these variables are used in the assertions that follow. How about this one?